### PR TITLE
C-2622: Add automations API

### PR DESCRIPTION
**Pull request by tk26**
## Description of the change

> Adds Automations API

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> None

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Review by cgradwohl — APPROVED** [review]

**Review by aydrian — APPROVED** [review]

File: src/automations/types.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+export interface IAutomation {
+  cancelation_token?: string;
+  steps: any[];
```
Comment (suhasdeshpande) [inline src/automations/types.ts:3]: could we be more explicit here?

File: src/automations/types.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+export interface ICourierAutomationInvokeParams {
+  brand?: string;
+  data?: object;
+  profile?: object;
```
Comment (suhasdeshpande) [inline src/automations/types.ts:9]: prefer `Record<string, unknown` instead of object

**Review by suhasdeshpande — CHANGES_REQUESTED** [review]
Some minor changes to types but seems fine otherwise.

File: src/automations/types.ts (hunk 1 above)
Comment (tk26) [reply src/automations/types.ts]: added types for all types of step we support. No pun intended :D

File: src/automations/types.ts [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+  override?: object;
+  profile?: object;
+  recipient?: string;
+  template?: string;
+}
+
+export interface IAutomationSendListStep extends IAutomationStep {
+  action: "send-list";
+  brand?: string;
+  data?: object;
```
Comment (suhasdeshpande) [inline src/automations/types.ts:33]: let's replace object with `Record<string, unknown>` object is still too big of type and potentially can hold almost anything that can be serializable?

**Review by aydrian — APPROVED** [review]

File: src/__tests__/automations.test.ts [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      })
+    ).resolves.toMatchObject(mockAutomationInvokeResponse);
+  });
+
+  test(".invokeAutomationTemplate", async () => {
+    const { automations } = CourierClient({
+      authorizationToken: "AUTH_TOKEN"
+    });
+
+    await expect(
```
Comment (suhasdeshpande) [inline src/__tests__/automations.test.ts:65]: Are there any error situation that are worth putting in unit tests?

**Review by suhasdeshpande — APPROVED** [review]
Some non-blocking question or nice to have request

- Would be nice to document what failure scenarios would look like and document those in unit tests. 
- Unit tests geared around some real world use-case of why one would use automation would be extremely nice as we start requesting more folks to use or try this out.

File: src/__tests__/automations.test.ts (hunk 4 above)
Comment (tk26) [reply src/__tests__/automations.test.ts:65]: Not really. We throw a 400 if required attributes aren't passed along which TS would take care of, and 5XX would probably be in axios error handling spectrum.

**Pull request merged**